File: benchmarks/tsb/bench_drop_duplicates_fn.ts
/**
 * Benchmark: dropDuplicatesSeries / dropDuplicatesDataFrame — standalone drop-duplicates on 100k elements.
 * Outputs JSON: {"function": "drop_duplicates_fn", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series, DataFrame, dropDuplicatesSeries, dropDuplicatesDataFrame } from "../../src/index.ts";

const SIZE = 100_000;
const WARMUP = 5;
const ITERATIONS = 50;

const s = new Series({ data: Array.from({ length: SIZE }, (_, i) => i % 1000) });
const df = new DataFrame({
  a: Array.from({ length: SIZE }, (_, i) => i % 1000),
  b: Array.from({ length: SIZE }, (_, i) => i % 500),
});

for (let i = 0; i < WARMUP; i++) {
  dropDuplicatesSeries(s);
  dropDuplicatesDataFrame(df);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  dropDuplicatesSeries(s);
  dropDuplicatesDataFrame(df);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "drop_duplicates_fn",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_drop_duplicates_fn.py
"""
Benchmark: Series.drop_duplicates / DataFrame.drop_duplicates on 100k elements.
Mirrors dropDuplicatesSeries / dropDuplicatesDataFrame standalone functions.
Outputs JSON: {"function": "drop_duplicates_fn", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""
import json
import time
import numpy as np
import pandas as pd

SIZE = 100_000
WARMUP = 5
ITERATIONS = 50

s = pd.Series(np.arange(SIZE) % 1000)
df = pd.DataFrame({
    "a": np.arange(SIZE) % 1000,
    "b": np.arange(SIZE) % 500,
})

for _ in range(WARMUP):
    s.drop_duplicates()
    df.drop_duplicates()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.drop_duplicates()
    df.drop_duplicates()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "drop_duplicates_fn",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
